### PR TITLE
A few cleanups and minor improvements to save_analysis

### PR DESCRIPTION
**Pull request by ljedrz**
- calculate the capacity of some `Vec`s
- change`to_owned()` to `clone()` for the purposes of `lower_attributes`
- remove a superfluous `clone()`
- prefer `to_owned()` to `to_string()`
- a few other minor improvements

### CODE REVIEW

File: src/librustc_save_analysis/dump_visitor.rs [hunk 1]
```diff
@@ -1608,8 +1605,7 @@ impl<'l, 'tcx: 'l, 'll, O: DumpOutput + 'll> Visitor<'l> for DumpVisitor<'l, 'tc
                 }
             }
             ast::ExprKind::Closure(_, _, _, ref decl, ref body, _fn_decl_span) => {
-                let mut id = String::from("$");
-                id.push_str(&ex.id.to_string());
+                let id = format!("${}", &ex.id);
```
Comment (oli-obk) [inline src/librustc_save_analysis/dump_visitor.rs]: the ampersand is unecessary

File: src/librustc_save_analysis/lib.rs [hunk 2]
```diff
@@ -210,7 +210,7 @@ impl<'l, 'tcx: 'l> SaveContext<'l, 'tcx> {
                     decl_id: None,
                     docs: self.docs_for_attrs(&item.attrs),
                     sig: sig::item_signature(item, self),
-                    attributes: lower_attributes(item.attrs.clone(), self),
+                    attributes: lower_attributes(item.attrs.to_owned(), self),
```
Comment (oli-obk) [inline src/librustc_save_analysis/lib.rs]: seems weird to go from `clone` to `to_owned`. One wouldn't call `to_owned` on a `&String`, only on a `&str` I think?

**Review by oli-obk — CHANGES_REQUESTED** [review]

File: src/librustc_save_analysis/lib.rs (hunk 2 above)
Comment (ljedrz) [reply src/librustc_save_analysis/lib.rs]: Both variants are currently present in the file and I thought that `to_owned()` would be preferable here, but I don't mind performing the change the other way around.

**Comment by ljedrz** [comment]
Thanks for the review, comments addressed.

**Review by oli-obk — APPROVED** [review]

**Pull request merged**